bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/api/EnedisHttpApi.java

```diff
@@ -216,45 +129,109 @@ private String getData(String url) throws LinkyException {
     }
 
     public PrmInfo getPrmInfo() throws LinkyException {
+        PrmInfo result = new PrmInfo();
+        Customer customer = getCustomer();
+        UsagePoint usagePoint = customer.usagePoints[0];
+
+        result.contractInfo = usagePoint.contracts;
+        result.usagePointInfo = usagePoint.usagePoint;
+        result.identityInfo = getIdentity();
+        result.addressInfo = getAddress();
+        result.contactInfo = getContact();
+
+        result.prmId = result.usagePointInfo.usagePointId;
+        result.customerId = customer.customerId;
+
+        return result;
+    }
+
+    public Customer getCustomer() throws LinkyException {
+        if (!connected) {
+            initialize();
+        }
+        String data = getData(String.format("%s/%s/cache", CONTRACT_URL, config.prmId));
+        if (data.isEmpty()) {
+            throw new LinkyException("Requesting '%s' returned an empty response", CONTRACT_URL);
+        }
+        try {
+            CustomerReponse cResponse = gson.fromJson(data, CustomerReponse.class);
+            if (cResponse == null) {
+                throw new LinkyException("Invalid customer data received");
+            }
+            return cResponse.customer;
+        } catch (JsonSyntaxException e) {
+            logger.debug("invalid JSON response not matching PrmInfo[].class: {}", data);
+            throw new LinkyException(e, "Requesting '%s' returned an invalid JSON response", CONTRACT_URL);
+        }
+    }
+
+    public AddressInfo getAddress() throws LinkyException {
         if (!connected) {
             initialize();
         }
-        String data = getData(PRM_INFO_URL);
+        String data = getData(String.format("%s/%s/cache", ADDRESS_URL, config.prmId));
```

Comment:
You could use `"%s/%s/cache".formatted(ADDRESS_URL, config.prmId)`

Reply:
fix in next commit